indexing-service/src/main/java/org/apache/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -219,6 +220,23 @@ public TaskStatus call()
                           );
                         }
 
+                        // Override task specific javaOptsArray
+                        try {
+                          List<String> taskJavaOptsArray = jsonMapper.convertValue(
+                              task.getContextValue(ForkingTaskRunnerConfig.JAVA_OPTS_ARRAY_PROPERTY),
+                              new TypeReference<List<String>>() {}
+                          );
+                          if (taskJavaOptsArray != null) {
+                            Iterables.addAll(command, taskJavaOptsArray);
+                          }
+                        }
+                        catch (Exception e) {
+                          throw new IllegalArgumentException(
+                              "javaOptsArray in context of task: " + task.getId() + " must be an array of strings.",
```

Comment:
should be `ForkingTaskRunnerConfig.JAVA_OPTS_ARRAY_PROPERTY` value instead of `javaOptsArray` - also no need to add the exception we got while parsing.

Reply:
Done